homeassistant/components/sms/__init__.py

```diff
@@ -1,27 +1,40 @@
 """The sms component."""
-import voluptuous as vol
+import logging
+
+import voluptuous as vol  # pylint: disable=import-error
 
 from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
 from homeassistant.const import CONF_DEVICE, Platform
 from homeassistant.core import HomeAssistant
 from homeassistant.helpers import config_validation as cv
 from homeassistant.helpers.typing import ConfigType
 
-from .const import DOMAIN, SMS_GATEWAY
+from .const import CONF_BAUD_SPEED, DEFAULT_BAUD_SPEED, DOMAIN, SMS_GATEWAY
 from .gateway import create_sms_gateway
 
+_LOGGER = logging.getLogger(__name__)
+
 PLATFORMS = [Platform.SENSOR]
 
 CONFIG_SCHEMA = vol.Schema(
-    {DOMAIN: vol.Schema({vol.Required(CONF_DEVICE): cv.isdevice})},
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Required(CONF_DEVICE): cv.isdevice,
+                vol.Optional(
+                    CONF_BAUD_SPEED, default=DEFAULT_BAUD_SPEED
+                ): cv.positive_int,
```

Comment:
For integrations that connect to devices or services, we no longer accept new YAML configuration or changes.

This integration needs to be refactored first to use a configuration flow and config entries first.

More information about this can be found in Architecture Decision Record:

- ADR-0010: <https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md#decision>

Since this integration already has configuration flow support, the YAML setup should have been deprecated.

Reply:
How does the yaml setup gets deprecated?

Reply:
I figure it out. yaml config is now deprecated.